src/opendr/perception/object_detection_2d/nanodet/nanodet_learner.py

```diff
@@ -385,117 +410,189 @@ def _save_onnx(self, onnx_path, do_constant_folding=False, verbose=True, conf_th
-                                             "conf_threshold": conf_threshold, "iou_threshold": iou_threshold}}
-            model_traced.save(export_path)
+        metadata = {"model_paths": ["nanodet_{}.pth".format(self.cfg.check_point_name)], "framework": "pytorch",
+                    "format": "pth", "has_data": False, "optimized": True, "optimizer_info": {},
+                    "inference_params": {"input_size": self.cfg.data.val.input_size, "classes": self.classes,
+                                         "conf_threshold": predictor.conf_thresh,
+                                         "iou_threshold": predictor.iou_thresh}}
+        model_traced.save(export_path)
 
-            with open(os.path.join(jit_path, "nanodet_{}.json".format(self.cfg.check_point_name)),
-                      'w', encoding='utf-8') as f:
-                json.dump(metadata, f, ensure_ascii=False, indent=4)
+        with open(os.path.join(jit_path, "nanodet_{}.json".format(self.cfg.check_point_name)),
+                  'w', encoding='utf-8') as f:
+            json.dump(metadata, f, ensure_ascii=False, indent=4)
 
-            if verbose:
-                print("Finished export to TorchScript.")
+        self._info("Finished export to TorchScript.", verbose)
 
     def _load_jit(self, jit_path, verbose=True):
-        if verbose:
-            print("Loading JIT model from {}.".format(jit_path))
-
+        jit_path = jit_path[0]
+        self._info(f"Loading JIT model from {jit_path}.", verbose)
         self.jit_model = torch.jit.load(jit_path, map_location=self.device)
 
     def optimize(self, export_path, verbose=True, optimization="jit", conf_threshold=0.35, iou_threshold=0.6,
-                 nms_max_num=100):
+                 nms_max_num=100, hf=False, dynamic=False, ch_l=False, lazy_load=True):
         """
-        Method for optimizing the model with ONNX or JIT.
+        Method for optimizing the model with ONNX, JIT or TensorRT.
         :param export_path: The file path to the folder where the optimized model will be saved. If a model already
         exists at this path, it will be overwritten.
         :type export_path: str
         :param verbose: if set to True, additional information is printed to STDOUT
         :type verbose: bool, optional
-        :param optimization: the kind of optimization you want to perform [jit, onnx]
+        :param optimization: the kind of optimization you want to perform [jit, onnx, trt]
         :type optimization: str
         :param conf_threshold: confidence threshold
         :type conf_threshold: float, optional
         :param iou_threshold: iou threshold
         :type iou_threshold: float, optional
         :param nms_max_num: determines the maximum number of bounding boxes that will be retained following the nms.
-        :type nms_max_num: int
+        :type nms_max_num: int, optional
+        :param hf: determines model's floating point precision.
```

Comment:
Similar to suggestions in the docs.

```suggestion
        :param hf: determines if half precision is used.
```